opensearch/services/datasets/datasets.php

```diff
@@ -351,6 +387,7 @@ public function getQueryBody($type) {
 				$body .= "OPTIONAL { ?dataset dco:associatedDCOPortalGroup ?gp . ?gp rdfs:label ?g_l . } ";
 				$body .= "OPTIONAL { ?project dco:relatedDataset ?dataset ; rdfs:label ?pl . } ";
 				$body .= "OPTIONAL { ?dataset obo:ERO_0000045 ?acc . } ";
+				$body .= "OPTIONAL { ?dataset dco:hasDataType ?d . ?d rdfs:label ?d_l . }";
```

Comment:
If you are using the getDatatypesByDataset function to execute a separate query to get datatypes for the dataset then you do not need to include the optional query constraint on the main dataset query for the dataType_label.